bin/sce_qc_report.R

```diff
@@ -17,39 +18,120 @@ option_list <- list(
   ),
+  make_option(
+    opt_str = c("-l", "--library_id"),
+    type = "character",
+    help = "Library identifier for report"
+  ),
   make_option(
     opt_str = c("-s", "--sample_id"),
     type = "character",
-    help = "Sample identifier for report"
+    help = "Sample identifier for metadata file"
   ),
   make_option(
-    opt_str = c("-o", "--output_file"),
+    opt_str = c("-q", "--qc_report_file"),
     default = "qc_report.html",
     type = "character",
-    help = "path to output file"
+    help = "path to QC report output file"
+  ),
+  make_option(
+    opt_str = "--metadata_csv",
+    default = "metadata.csv",
+    type = "character",
+    help = "path to metadata csv output file"
+  ),
+  make_option(
+    opt_str = "--metadata_json",
+    default = "metadata.json",
+    type = "character",
+    help = "path to metadata json output file"
+  ),
+  make_option(
+    opt_str = "--genome_assembly",
+    type = "character",
+    default = NA,
+    help = "workflow github url"
+  ),
+  make_option(
+    opt_str = "--workflow_url",
+    type = "character",
+    default = NA,
+    help = "workflow github url"
+  ),
+  make_option(
+    opt_str = "--workflow_version",
+    type = "character",
+    default = NA,
+    help = "workflow version identifier"
+  ),
+  make_option(
+    opt_str = "--workflow_commit",
+    type = "character",
+    default = NA,
+    help = "workflow commit hash"
   )
+
 )
 
 opt <- parse_args(OptionParser(option_list = option_list))
-if(is.null(opt$sample_id)){
-  stop("A `sample_id` is required.")
+if(is.null(opt$library_id)){
+  stop("A `library_id` is required.")
 }
-# check that input files exists
+# check that input files exist
 if(is.null(opt$unfiltered_sce) || !file.exists(opt$unfiltered_sce)){
   stop("Unfiltered .rds file missing or `unfiltered_sce` not specified.")
 }
 if(is.null(opt$filtered_sce) || !file.exists(opt$filtered_sce)){
   stop("Filtered .rds file missing or `filtered_sce` not specified.")
 }
 
+if (opt$workflow_url == "null"){
+  opt$workflow_url <- NA
+}
+if (opt$workflow_version == "null"){
+  opt$workflow_version <- NA
+}
+if (opt$workflow_commit == "null"){
+  opt$workflow_commit <- NA
+}
 
 # read sce files
 unfiltered_sce <- readr::read_rds(opt$unfiltered_sce)
 filtered_sce <- readr::read_rds(opt$filtered_sce)
 
+# Compile metadata for output files
+sce_meta <- metadata(unfiltered_sce)
+
+# check for alt experiments (CITE-seq, etc)
+alt_expts <- altExpNames(unfiltered_sce)
+has_citeseq <- "CITEseq" %in% alt_expts
+
+metadata_list <- list(
+  library_id = opt$library_id,
+  sample_id = opt$sample_id,
+  date_processed = lubridate::now(),
+  filtered_cells = ncol(filtered_sce),
+  unfiltered_cells = ncol(unfiltered_sce),
+  total_reads = sce_meta$total_reads,
+  mapped_reads = sce_meta$mapped_reads,
+  genome_assembly = opt$genome_assembly,
+  mapping_index = sce_meta$reference_index,
+  transcript_type = sce_meta$transcript_type,
```

Comment:
Do we want to add in the salmon and alevin fry versions here? I feel like if we are already outputting this file with the index, assembly, and transcript information it would be helpful to include as much information that we have about the sample. It could come in handy and depending on if we include this file with the download or not, I think it could be useful for people to have.  
```suggestion
  transcript_type = sce_meta$transcript_type,
  salmon_version = sce_meta$salmon_version,
  alevin_fry_version = sce_meta$alevinfry_version,
```